src/faultlog/guard_with_eid_records.cpp

```diff
@@ -58,25 +60,136 @@ static int getGuardedTarget(struct pdbg_target* target, void* priv)
+
+            method.append(static_cast<uint32_t>(elem.elogId));
+            auto resp = bus.call(method);
+            resp.read(bmcLogId);
+        }
+        catch (const sdbusplus::exception::SdBusError& ex)
+        {
+            dbusErrorObjFound = false;
+            lg2::info(
+                "PEL might be deleted but guard entry is around {ELOG_ID)",
+                "ELOG_ID", elem.elogId);
+        }
+        uint32_t plid = 0;
+        if (dbusErrorObjFound)
+        {
+            Properties pelEntryProp;
+            std::string objPath = "/xyz/openbmc_project/logging/entry/" +
+                                  std::to_string(bmcLogId);
+            auto pelEntryMethod = bus.new_method_call(
+                "xyz.openbmc_project.Logging", objPath.c_str(),
+                "org.freedesktop.DBus.Properties", "GetAll");
+            pelEntryMethod.append("org.open_power.Logging.PEL.Entry");
+            auto pelEntryReply = bus.call(pelEntryMethod);
+            pelEntryReply.read(pelEntryProp);
+            for (const auto& [prop, propValue] : pelEntryProp)
+            {
+                if (prop == "PlatformLogID")
+                {
+                    auto plidPtr = std::get_if<uint32_t>(&propValue);
+                    if (plidPtr != nullptr)
+                    {
+                        plid = *plidPtr;
+                    }
+                }
+            }
+            // A single PEL could capture multiple guard records callouts so
+            // ignore a guard record if already cpatured as part of a PEL
+            if (std::find(liProcessedPels.begin(), liProcessedPels.end(),
+                          plid) != liProcessedPels.end())
+            {
+                lg2::info("Ignoring PEL as it has been already processed with "
+                          "another guard record {PEL_ID}",
+                          "PEL_ID", plid);
+                continue;
+            }
+        }
+        // D-Bus error object might be deleted
+        else
+        {
+            // hwas state will be updated only during reipl till then plid will
+            // be zero, if zero assume it as new serviceable event else check if
+            // it is already processed
+            plid = static_cast<uint32_t>(hwasState.deconfiguredByEid);
+            if (plid != 0)
+            {
+                // A single PEL could capture multiple guard records callouts so
+                // ignore a guard record if already cpatured as part of a PEL
+                if (std::find(liProcessedPels.begin(), liProcessedPels.end(),
+                              plid) != liProcessedPels.end())
+                {
+                    lg2::info("Ignoring PEL as it has been already processed "
+                              "with another guard record {PEL_ID}",
+                              "PEL_ID", plid);
+                    continue;
+                }
+            }
+        }
+        liProcessedPels.push_back(plid);
```

Comment:
do we need to check the plid != 0? because if ln no 167 has failed, it will come here and add a 0

Reply:
I am using vector so that I can add duplicate entries, assuming there are 3 guard records and its 3 error objects are deleted, all of them will have plid as zero till next reipl, but still I need to log 3 nag entries in the json file, so need to return count as 3.